drivers/sensor/mchp_tach_xec/tach_mchp_xec.c

```diff
@@ -8,7 +8,9 @@
 #define DT_DRV_COMPAT microchip_xec_tach
 
 #include <errno.h>
+#include <zephyr/kernel.h>
 #include <zephyr/device.h>
+#include <zephyr/sys/sys_io.h>
```

Comment:
you're probably looking for `zephyr/arch/cpu.h` instead, this header doesn't expose the `sys_*` APIs (weird, but that's how it is).

Reply:
yes, this is very confusing and I noticed that, we need to fix this.

Reply:
huh, the cpu.h loads arch.h which then includes <zephyr/arch/common/sys_io.h>